### ISSUE TITLE
Confused About How The Mod Works

### ISSUE THREAD

**Issue by AdamGaffney96** [Aragami-delp/PlateUp_Mods#6]
Hi there,

I'm writing it here as I'm unsure if it's an issue with my install or if I'm just not getting how the mod is meant to work, however I can't seem to have more than one save. I made a save on a custom map and named this Pizza Test. This showed up fine in my save games. I then used the "Return to lobby" button and could restore it from the "Restore Save" button on the ground. I then started a new run with a different menu on a different map, however when I went to "Save System" to save this one as a new save, all I got was the "Already Saved" and "Back" buttons, with no option to Save as new. From here if I return to the lobby and restore save it is my new one, however if I go to "Pizza Test" and hit "Load" and confirm it, then it's still the new one upon trying to Restore Save. 

However then when I fully quit the game and restart, then hit "Restore Save" on the ground without doing any loading, it's my original "Pizza Test" save and the new one was lost? I'm unsure as to why I seem unable to create a new save which makes me think I must be doing something wrong.

Note that I have BepinEx, Better Display Stands, Kitchen Designer, Prep Ghost and Save System all installed but not sure if it could be a compatibility issue. I can provide any extra info that could help upon request, also happy to try record video if it would help, thanks.

**EDIT: Adding new info after extra testing**

So I've found that if I abandon instead of returning to lobby, then the next game I start I can add a new save. However when abandoning this removed the "Recover Save" thing on the ground, and pressing "Load" then "Confirm Loading" on a save does absolutely nothing, the recover save button still isn't there and I can't actually load into the save. If I don't abandon, the "Recover Save" is there however I always just load into the new map I started, and cannot load the saves unless I fully quit the game first.

**Comment by AdamGaffney96** [comment]
I created a video to hopefully demo some of the issues I'm having and added chapters on the description, hopefully this can help illuminate if I'm doing something wrong or if there's an actual issue with the mod.
https://www.youtube.com/watch?v=HS7uNRw-Ivk

**Comment by Aragami-delp** [comment]
Thank you for reporting, this is a bug reported by multiple people, but I've never been able to reproduce it. Your description looks very detailed. I have an potential fix and see if it works and I plan to push an update later today, when I get back home.

**Comment by AdamGaffney96** [comment]
> Thank you for reporting, this is a bug reported by multiple people, but I've never been able to reproduce it. Your description looks very detailed. I have an potential fix and see if it works and I plan to push an update later today, when I get back home.

Excellent news, hopefully all the detail can help get to the bottom of it! Happy to do some testing later today if you have a build and see if it fixes the issue.

**Comment by AdamGaffney96** [comment]
Apologies, but I just updated to 1.3.2 to test it out. I had a save game called "Multiplayer" and decided to start a new game for my single player runs. I completed day 1 on a new map, went to the save section and had a "Save Now" button so I thought great, it's working now. I hit "Save Now" and never got the option to name, it just saved. It turns out it's overwritten the "Multiplayer" save I had despite being a different map, recipe and starting afresh after having restarted the game. I'm not sure if this is possible and I'm sure any changes would be a lot of work, but would there be a way to have "New Save" and "Update Existing Save" instead or something like that? It's hard to tell if you're going to overwrite or create a new save until it's already too late and you've overwritten a good save.

I want to say I absolutely appreciate what you're doing as this straight up just should be a feature built into the game, and this is something desperately needed in this game. I just want to make sure that I'm doing this right and that I am able to use this properly. Can provide any extra detail if necessary and can split into a different issue report if that's better also.

**Comment by AdamGaffney96** [comment]
I will also note:- I was able to recover my save thankfully by just going in manually to the `LocalLow` folder and finding that instead of overwriting my save it'd created a new file in the same folder, so there was essentially two different world saves in one folder, and it was only accessing the most recent one. I manually created a folder and moved the single player game into this folder, then updated the SaveInfo.json to include the required info to be picked up by the mod. Then when I went into the game I had two saves as expected. 

However what this essentially means is that to create a new save, I have to manually create the folder and update the SaveInfo.json as it doesn't seem to be possible for me in-game. Happy to do this for the moment as at least it works and is still way better than having my game unplayable for a week until me and my friend can play again, however thought it's best to raise just in case this is unexpected. Can make a demo tomorrow if necessary.
